#! /bin/sh /usr/share/dpatch/dpatch-run
## r221 | relnev | 2004-11-06 09:26:11 +0100 (Sa, 06 Nov 2004) | 2 lines
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fixed an incorrect array access.

@DPATCH@
diff -urNad rott-1.0+dfsg~/rt_game.c rott-1.0+dfsg/rt_game.c
--- rott-1.0+dfsg~/rt_game.c	2008-01-10 09:22:34.000000000 +0100
+++ rott-1.0+dfsg/rt_game.c	2008-01-10 09:22:36.000000000 +0100
@@ -104,12 +104,12 @@
 
 static int KeyX[4]  = {KEY1_X, KEY2_X, KEY3_X, KEY4_X};
 
-static char *Names[ 5 ] =
+static const char *Names[ 5 ] =
    {
    "Taradino", "Thi",     "Doug",  "Lorelei", "Ian Paul"
    };
 
-static char *LastNames[ 5 ] =
+static const char *LastNames[ 5 ] =
    {
    "Cassatt",  "Barrett", "Wendt", "Ni",      "Freeley"
    };
@@ -487,7 +487,7 @@
       VW_MeasurePropString( LastNames[ character ], &width, &height );
 
       DrawGameString ( MEN_X + 44 - width, MEN_Y + 8,
-         Names[ character + 5 ], bufferofsonly );
+         LastNames[ character ], bufferofsonly );
 
       UpdateLives( locplayerstate->lives );
       UpdateScore( gamestate.score );
@@ -996,7 +996,7 @@
 //
 //******************************************************************************
 
-void DrawGameString (int x, int y, char * str, boolean bufferofsonly)
+void DrawGameString (int x, int y, const char * str, boolean bufferofsonly)
 {
    byte* tempbuf;
 
diff -urNad rott-1.0+dfsg~/rt_game.h rott-1.0+dfsg/rt_game.h
--- rott-1.0+dfsg~/rt_game.h	2008-01-09 13:52:27.000000000 +0100
+++ rott-1.0+dfsg/rt_game.h	2008-01-10 09:22:36.000000000 +0100
@@ -93,7 +93,7 @@
 
 void DrawKills (boolean bufferofsonly);
 void DrawPlayers ( void );
-void DrawGameString (int x, int y, char * str, boolean bufferofsonly);
+void DrawGameString (int x, int y, const char * str, boolean bufferofsonly);
 void DrawNumber (int x, int y, int width, int which, boolean bufferofsonly);
 void TakeDamage (int points, objtype *attacker);
 void HealPlayer (int points, objtype * ob);
